Source/Core/Core/IOS/Network/IP/Top.cpp

```diff
@@ -857,12 +861,25 @@ IPCCommandResult NetIPTop::HandleGetInterfaceOptRequest(const IOCtlVRequest& req
         FREE(AdapterAddresses);
       }
     }
+#elif defined(__linux__) && !defined(__ANDROID__)
+    int error = res_init();
+    if (!error)
+    {
+      address = ntohl(_res.nsaddr_list[0].sin_addr.s_addr);
+    }
+    else
+    {
+      WARN_LOG(IOS_NET, "Call to res_init failed, falling back to defaults: %X", error);
```

Comment:
This message isn't very clear. `error` isn't the defaults we're using as fallback, it would be nice to mention them (`8.8.8.8` and `8.8.4.4`).

Then, according to the manual, the only return values possibles are `0` and `-1`. Giving the error code won't help that much and on top of that some implementations seem to set `errno` while some don't.

Feel free to replace the condition with `if (res_init() != 0)` if you want to.

Reply:
Fixed